core/storage/user/gae_models.py

```diff
@@ -120,6 +120,13 @@ class UserSettingsModel(base_models.BaseModel):
     # May be None.
     first_contribution_msec = datastore_services.FloatProperty(default=None)
 
+    # TODO(#12755): Currently, "roles" and "banned" fields are not in use. These
+    # new fileds will be used once their values are populated.
+    # A list of roles assigned to the user.
+    roles = datastore_services.StringProperty(repeated=True, indexed=True)
```

Comment:
choices?

Reply:
Done! (Also added a test to ensure all the allowed roles are used in roles_services and actions defined to them)